conda_build/skeletons/cran.py

```diff
@@ -644,14 +645,22 @@ def get_cran_archive_versions(cran_url, session, package, verbose=True):
 def get_cran_index(cran_url, session, verbose=True):
     if verbose:
         print("Fetching main index from %s" % cran_url)
-    r = session.get(cran_url + "/src/contrib/")
+    r = session.get(cran_url + "/src/contrib/PACKAGES.gz")
     r.raise_for_status()
     records = {}
-    for p in re.findall(r'<td><a href="([^"]+)">\1</a></td>', r.text):
-        if p.endswith('.tar.gz') and '_' in p:
-            name, version = p.rsplit('.', 2)[0].split('_', 1)
-            records[name.lower()] = (name, version)
+    package = None
+    for line in gzip.decompress(r.content).decode('utf-8', errors='replace').splitlines():
+        if package is None:
+            if line.startswith('Package: '):
+                package = line.rstrip().split(' ', 1)[1]
+        elif line.startswith('Version: '):
+            records[package.lower()] = (package, line.rstrip().split(' ', 1)[1])
+            package = None
     r = session.get(cran_url + "/src/contrib/Archive/")
+    if r.status_code in (403, 404):
+        if verbose:
+            print("Cannot fetch an archive index from %s" % cran_url)
```

Comment:
I think it would be better to use
```suggestion
            warnings.warn("Cannot fetch an archive index from %s" % cran_url, UserWarning)
```
and you will need to import 
```python
import warnings
```

Reply:
Just one more thing, do you think you can create a test to cover this scenario please? That would be awesome!

Reply:
Will work on it!

Reply:
I'm not sure I agree with the use of `warnings.warn` here. I really would consider this more informational than a warning. Some sources of R packages just don't have an Archive directory, _by design_. And if the Archive directory is absent it should be treated as if it was there, but empty.

Put another way, this module is screaming for a proper `logging` implementation—and this would be `logging.INFO` or maybe even `logging.DEBUG`, not `logging.WARN`.

Reply:
Here's an example of a source of R packages with no Archive directory—http://datacube.wu.ac.at/